Establish the connection between the frond end back end for the SIGN-UP route.

UserOlena opened the pull request:
*   Implemented field validation for the username, email, and password inputs.
 *   Added an event listener to track changes in the input fields and enable the "Create Account" button only if all fields pass the validation. Otherwise, the button remains disabled.
 *  Implemented a redirection to the home page upon successful sign-up.
 *   Added user_id and logged_in status to the req.session for user authentication.

To-do:

 *   Implement functionality on the client side to display error messages from the backend if the email or username already exists in the database.
 *   Update the styling of the "Create Account" button to visually indicate whether it is enabled or disabled.
 *   Consider adding an additional field to the sign-up form to accommodate any new columns added to the user model that need to be filled.
* maybe something else i cannot think about right now.

p.s.
The functionality is operational, and new users have been successfully added to the database. However, it is possible that certain updates or modifications will be required in the future to enhance the app further.

bbandhu commented on public/handlers/user-auth.js:
```diff
@@ -0,0 +1,127 @@
+        password: $signupPassword.val().trim(),
+    }
+
+    const response = await fetch('http://localhost:3001/api/user/signup', {
```
Can we update the path to /api/user/signup instead of prepending localhost?

UserOlena replied:
Yes. you right! Initially I had it like  /api/user/signup, but since I struggled with errors (because I had no idea I would need to include preventDafault), I tried different ways to resolve it and once included preventDefault, forgot to change the URL. Just changed and tested it. It works the same with  /api/user/signup and I committed those changes

bbandhu reviewed (commented):
On line 18 - Can we update the path to /api/user/signup instead prepending localhost?

JDempe commented:
![image](https://github.com/JDempe/project2/assets/123279032/cb4a1568-fe72-492e-997c-ae2a989abefb)

The validation for the password, is there a certain amount of alphanumeric / special characters needed?  I have uppercase, lowercase, number, and special and it is giving me the error, preventing sign up.

UserOlena commented:
> 

Regarding the validation of passwords, it is required to have at least one numerical, one uppercase, and one special symbol character. I tested the functionality by pulling the main branch and loading my branch on top of it. The validation works well, and the user was saved to the database if at least one of the mentioned characters was included. The sign-up fields are validated when the user clicks out of the field and moves to another one. If the validation criteria are not met, an error message will appear. However, the error message does not mention the requirement for an uppercase character. Initially, I forgot to include the validation for an uppercase character. After I added the uppercase validation, I forgot to update the error message. Just Updated it and committed the changes. I am including a screenshot of the sign-up form that passed the validation and saved the user to the database.

<img width="596" alt="Screenshot 2023-05-17 at 8 37 32 AM (2)" src="https://github.com/JDempe/project2/assets/122197592/f67fd64e-e1f7-43c7-8180-c67b73a65654">

JDempe commented:
Okay, it does seem to be the exclamation point then.  I tested it further and it works great with at least one of each, it just doesn't accept ! as a special character.  The first image fails, the second image passes. 
![image](https://github.com/JDempe/project2/assets/123279032/34c480d6-98b9-4845-b5b7-5945f5f70166)
![image](https://github.com/JDempe/project2/assets/123279032/ec05d1e9-8976-458d-b38c-639a263b2b42)

JDempe reviewed (approved):
Woo!
